java-frontend/src/main/java/org/sonar/plugins/java/api/JavaFileScanner.java

```diff
@@ -33,4 +33,16 @@ public interface JavaFileScanner extends JavaCheck {
    */
   void scanFile(JavaFileScannerContext context);
 
+  /**
+   * Scan based on the raw file and cached data (ie: No tree is available at this stage).
+   * The rule should leverage data from the read cache.
+   * The rule should persist data to the write cache for future analyses.
+   *
+   * @param inputFileScannerContext The file that will eventually be scanned
+   * @return True if successful (ie: no further scanning is required). False by default or if the file cannot be scanned exhaustively without contents.
+   */
+  @Beta
+  default boolean scanWithoutParsing(InputFileScannerContext inputFileScannerContext) {
+    return false;
```

Comment:
After discussing last week, we actually want to return `true` by default now. We should also update the javadocs of course.